plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -534,6 +536,19 @@ unsafeEvaluateCek
     -> EvaluationResult (Term Name uni fun ())
 unsafeEvaluateCek runtime = either throw id . extractEvaluationResult . evaluateCek runtime
 
+-- | This is like unsafeEvaluateCek, but it takes an initial budget and returns the final budget
```

Comment:
We've got rather a lot of evaluation functions in here now.  We should maybe try to rationalise them at some point (but not right now!).